src/main/kotlin/com/terraformation/backend/file/ThumbnailStore.kt

```diff
@@ -122,49 +131,58 @@ class ThumbnailStore(
     val filesRow = filesDao.fetchOneById(fileId) ?: throw FileNotFoundException(fileId)
     val photoUrl = filesRow.storageUrl!!
 
-    val resizedImage = scalePhoto(photoUrl, maxWidth, maxHeight)
-    val buffer = encodeAsJpeg(resizedImage)
-    val size = buffer.size
+    try {
+      if (!semaphore.tryAcquire(0, TimeUnit.SECONDS)) {
+        log.debug("Maximum number of thumbnails already being generated; waiting for one to finish")
+        semaphore.acquire()
```

Comment:
does this timeout? just curious..

Reply:
It doesn't, but that's a good point; I should make it time out.